movie-app/src/header/NavBar.js

```diff
@@ -1,26 +1,73 @@
-const NavBar = () => {
-  return (
-      <div className = "header">
+import { useState } from "react";
 
-        <nav className = "navigation">
+const NavBar = ({ movieSearch, setMovieSearch }) => {
+  const [searchOpen, setSearchOpen] = useState();
 
-          <div className= "logo-container">
-            <img src={'./logo.svg'} alt="ToadTv logo" className="logo"/>
-          </div>
+  const clearText = () => {
+    document.getElementById("input").value = "";
+  };
 
-          <div className ="searchbar-container">
-            <div className = "searchbar"></div>
-          </div>
+  return (
+    <div className="header">
+      <nav className="navigation">
+        <div className="logo-container">
+          <img src={"./logo.svg"} alt="ToadTv logo" className="logo" />
+        </div>
 
-        </nav>
+        <div>
+          <form
+            className={
+              searchOpen === undefined
+                ? "initial"
+                : searchOpen
+                ? `search-open`
+                : `search-close`
+            }
+            onSubmit={(e) => {
+              e.preventDefault();
+              if (e.target[0].value.length < 3) {
+                alert("Please search a movie title with more than two letters");
+              } else {
+                setMovieSearch(e.target[0].value);
+              }
+            }}
+          >
+            <img
+              onClick={() => {
+                setSearchOpen(true);
+              }}
+              className="search-icon"
+              src={"./Search_Icon.svg"}
+              alt="none"
+            />
+            <input
+              id="input"
+              type="text"
+              className={searchOpen ? "search-bar" : "search-bar-close"}
+            ></input>
+            <button
+              type="button"
+              onClick={(e) => {
+                setSearchOpen(false);
+                clearText();
+              }}
```

Comment:
I suggest writing this function outside of the JSX element for better readability.

Reply:
That is a good idea. I will do that now